rest-api-spec/src/main/resources/rest-api-spec/test/delete/70_mix_typeless_typeful.yml

```diff
@@ -0,0 +1,40 @@
+---
+"DELETE with typeless API on an index that has types":
+
+ - skip:
+      version: " - 6.99.99"
+      reason:  Typeless APIs were introduced in 7.0.0
+
+ - do:
+      indices.create: # not using include_type_name: false on purpose
+          index: index
+          body:
+              mappings:
+                  not_doc:
+                      properties:
+                          foo:
+                              type: "keyword"
+
+ - do:
+      index:
+          index:  index
+          type:   not_doc
+          id:     1
+          body:   { foo: bar }
+
+ - do:
+      catch:   bad_request
+      delete:
+          index:  index
+          type:   some_random_type
+          id:     1
+
+ - do:
+      delete:
+          index:  index
+          id:     1
+
+ - match:   { _index:   "index" }
+ - match:   { _type:    "_doc" }
+ - match:   { _id:      "1"}
+ - match:   { _version: 2}
```

Comment:
do we also need to check whether the doc gets deleted when specifying not_doc as a type? Or maybe that is already covered by existing tests?

Reply:
indeed this is tested by `delete/13_basic_with_types.yml`